### PR TITLE
Input challenges improvements

### PR DESCRIPTION
**Pull request by inyono**
Include the following issues:

* [x] #301
* [x] #302
* [x] #303
* [x] #312
* [x] #359

### CODE REVIEW

File: src/module/Ui/templates/entity/view/partials/text-exercise.twig [hunk 1]
```diff
@@ -31,7 +31,7 @@
     </section>
     {% if singleChoices|length > 0 %}
         <section>
-            <form class="single-choice-group">
+            <form class="single-choice-group {{ singleChoices|length>4 ? 'extended' }}">
```
Comment (inyono) [inline src/module/Ui/templates/entity/view/partials/text-exercise.twig]: @inyono: this shouldn't be there

**Comment by inyono** [comment]
@arekkas Ready for code review. After merging, I will add another PR that updates the language files.

**Comment by arekkas** [comment]
LGTM - could you rebase?

**Comment by inyono** [comment]
@arekkas done

**Pull request merged**